lib/charms/layer/azure.py

```diff
@@ -189,6 +193,18 @@ def send_additional_metadata(request):
         security_group_name=run_config.get("vnetSecurityGroup")
         if run_config.get("vnetSecurityGroup")
         else "juju-internal-nsg",
+        use_managed_identity=credentials.get("managed-identity")
+        if credentials.get("managed-identity", "") != "" 
+        else True,
+        aad_client=credentials.get("application-id") 
+        if credentials.get("application-id") 
+        else "",
+        aad_secret=credentials.get("application-password") 
+        if credentials.get("application-password") 
+        else "",
+        tenant_id=credentials.get("tenant-id") 
+        if credentials.get("tenant-id") 
+        else ""
```

Comment:
Upon further reflection, if the latter three of these fields are already part of the credentials provided, why do they need to also be sent separately over the relation?  Wouldn't it be sufficient to only add the `managed-identity` field?  Or, if that's also included in the credentials, then the interface wouldn't even need to change at all.

Reply:
>Oh, I think I'm starting to understand. So managed-identity=false basically means that instead of granting the integrator privileged permissions and trusting it to in turn only grant limited permissions onto the managed identities of the related units, you want to hand the integrator creds that you have manually limited and have it just act as a pass-through, right?

yes correct. The cloud provider instead of using the VM identity will just use the service principal.

Reply:
I think the changes suggested at the start of this thread are still worth applying, just for code clarity.

Reply:
got it, thanks!

Reply:
Removed the value checking @johnsca